web/api/v1/api.go

```diff
@@ -195,6 +195,11 @@ type API struct {
 	gatherer                  prometheus.Gatherer
 }
 
+type RankingFunction interface {
+	sum(v promql.Series) float64
+	avg(v promql.Series) float64
+}
```

Comment:
Interface should show a generic structure and should not know about specific implementations. What about this?
```suggestion
type RankingFunction interface {
	rank(v promql.Series) float64
}
```

Additionally, we could just have the below and ranking would be just a function.
```
type RankingFunction func(promql.Series) float64
```